tools/cockpit.spec

```diff
@@ -422,10 +426,7 @@ authentication via sssd/FreeIPA.
 %ghost %{_sharedstatedir}/selinux/%{selinuxtype}/active/modules/200/%{name}
 
 %pre ws
-getent group cockpit-ws >/dev/null || groupadd -r cockpit-ws
-getent passwd cockpit-ws >/dev/null || useradd -r -g cockpit-ws -d /nonexisting -s /sbin/nologin -c "User for cockpit web service" cockpit-ws
-getent group cockpit-wsinstance >/dev/null || groupadd -r cockpit-wsinstance
-getent passwd cockpit-wsinstance >/dev/null || useradd -r -g cockpit-wsinstance -d /nonexisting -s /sbin/nologin -c "User for cockpit-ws instances" cockpit-wsinstance
+%sysusers_create_compat %{SOURCE1}
```

Comment:
I wonder, how would that even work with this approach? Even if the sysusers file is taken from dist-git instead of upstream, it wuold still be shipped by the rpm and thus isn't yet available in `%pre`?

I suppose this will do that evil thing where it translates a sysusers file into `useradd` commands? I just found this:

```
❱❱❱ rpm -q --scripts  pipewire
preinstall scriptlet (using /bin/sh):

# generated from pipewire.sysusers
getent group 'pipewire' >/dev/null || groupadd -r 'pipewire' || :
getent passwd 'pipewire' >/dev/null || \
    useradd -r -g 'pipewire' -d '/run/pipewire' -s '/usr/sbin/nologin' -c 'PipeWire System Daemon' 'pipewire' || :
```

This is all new to me, so sorry for rambling.. But this looks hackish. I suppose it allows rpms to ship files which are owned by a sysusers owner. The only file in cockpit-ws which needs that is

```
%attr(4750, root, cockpit-wsinstance) %{_libexecdir}/cockpit-session
```

I think all this was the reason why in our initial approach we wanted to first get rid of that suid root binary (see PR #16808), and then there wouldn't be any need for static sysusers any more.

Reply:
Unfortunately I don't think it's possible to drop this copy due to how macro expansions work and until we fully have Fedora switch to using systemd sysusers (it's only using the format right now, not the command).

See also https://src.fedoraproject.org/rpms/systemd/pull-request/121 that might simplify things a bit.

Reply:
TBH I consider this is a blocker. This makes the packaging structure *more* complicated and brittle. So for me, it's either "chmod cockpit-session in `%post` as we do in Debian/Arch" (see [below](https://github.com/cockpit-project/cockpit/pull/20365#discussion_r1578998893)), or block this on getting rid of the suid root binary.